web-animations/timing-model/animations/updating-the-finished-state.html

```diff
@@ -0,0 +1,128 @@
+<!DOCTYPE html>
+<meta charset=utf-8>
+<title>Tests for updating the finished state of an animation</title>
+<link rel="help" href="https://w3c.github.io/web-animations/#updating-the-finished-state">
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script src="../../testcommon.js"></script>
+<body>
+<script>
+'use strict';
+
+function validateFinishEvent(t, event, animation) {
+  t.step(function() {
+    assert_equals(event.target, animation, "Animation should be event target");
+    assert_times_equal(event.currentTime, animation.currentTime, "Event currentTime should be animation currentTime");
+    assert_times_equal(event.timelineTime, animation.timeline.currentTime, "Event timelineTime should be animation timeline's currentTime");
```

Comment:
I'm not sure there is anything that guarantees this always holds. For one thing, I know that when we do the finish notification steps synchronously, the timeline time set on the event can end up being from the previous frame by the time the finish event handlers are run due to where we are in the event loop.